### PR TITLE
Add firewall hook

### PR DESCRIPTION
**Pull request by mafintosh**

### CODE REVIEW

File: binding.c [hunk 1]
```diff
@@ -32,7 +32,7 @@
     napi_fatal_exception(env, fatal_exception); \
     printf("oh no add that realloc\n"); \
   } else { \
-    napi_get_buffer_info(env, ret, (void **) &(self->read_buf), &(self->read_buf_free)); \
+    napi_get_buffer_info(env, res, (void **) &(self->read_buf), &(self->read_buf_free)); \
```
Comment (mafintosh) [inline binding.c:35]: unrelated typo (already worked by accident tho)

File: include/udx.h [hunk 2]
```diff
@@ -83,6 +83,7 @@ typedef void (*udx_socket_send_cb)(udx_socket_send_t *req, int status);
 typedef void (*udx_socket_recv_cb)(udx_socket_t *handle, ssize_t read_len, const uv_buf_t *buf, const struct sockaddr *from);
 typedef void (*udx_socket_close_cb)(udx_socket_t *handle);
 
+typedef int (*udx_stream_firewall_cb)(udx_stream_t *handle, const struct sockaddr *from);
```
Comment (kasperisager) [inline include/udx.h:86]: Do we need the socket as well?

Comment (mafintosh) [reply include/udx.h:86]: yes! handle, socket, from?

Comment (kasperisager) [reply include/udx.h:86]: I'd name it `socket` as it's a stream callback.

Comment (mafintosh) [reply include/udx.h:86]: ah haha, i meant 3 args total in that order yea? (handle, socket, from) 🤣

Comment (kasperisager) [reply include/udx.h:86]: Oh, yes, that's perfect 😂

**Review by kasperisager — APPROVED** [review]

**Pull request merged**